README.md

```diff
@@ -428,6 +437,203 @@ HTTP/1.1 200 OK
+{"cursor":{"partition":"0","offset":"545","event_type":"order.ORDER_RECEIVED","cursor_token":"bf6ee7a9-0fe5-4946-b6d6-30895baf0599"}}
+{"cursor":{"partition":"1","offset":"545","event_type":"order.ORDER_RECEIVED","cursor_token":"9ed8058a-95be-4611-a33d-f862d6dc4af5"}}
+```
+Each batch contains the following fields:
+
+- `cursor`: the cursor of the batch which should be used for committing the batch;
+- `events`: the array of events of this batch;
+- `info`: optional field that can hold some useful information (e.g. the reason why the stream was closed by Nakadi)
+
+Please also note that when stream is started the client receives a header `X-Nakadi-StreamId` which
+should be used when committing cursors.
+
+#### Subscription Cursors
+In Subscription API cursors have the following structure:
+```sh
+{
+  "partition": "5",
+  "offset": "543",
+  "event_type": "order.ORDER_RECEIVED",
+  "cursor_token": "b75c3102-98a4-4385-a5fd-b96f1d7872f2"
+}
+```
+Fields are:
+
+- `partition`: the partition this batch comes from;
+- `offset`: the offset of this batch; user should not do any operations with offset, for him it should be just a string;
+- `event_type`: specifies the event-type of the cursor (as in one stream there can be events of different event-types);
+- `cursor_token`: cursor token generated by Nakadi; useless for the user;
+
+#### Committing Cursors
+Cursors can be committed by posting to `/subscriptions/{subscriptionId}/cursors`. Example:
+```sh
+curl -v -X POST \
+    -H "Content-type: application/json" \
+    -H "X-Nakadi-StreamId: ae1e39c3-219d-49a9-b444-777b4b03e84c" \    
+    "http://localhost:8080/subscriptions/038fc871-1d2c-4e2e-aa29-1579e8f2e71f/cursors" \
+    -d '{
+      "items": [
+        {
+          "partition": "5",
+          "offset": "543",
+          "event_type": "order.ORDER_RECEIVED",
+          "cursor_token": "b75c3102-98a4-4385-a5fd-b96f1d7872f2"
+        },
+        {
+          "partition": "2",
+          "offset": "923",
+          "event_type": "order.ORDER_RECEIVED",
+          "cursor_token": "a28568a9-1ca0-4d9f-b519-dd6dd4b7a610"
+        }
+      ]
+    }'
+```
+Please be aware that `X-Nakadi-StreamId` header is required when doing a commit.
+The value should be the same as you get in `X-Nakadi-StreamId` header when opening a stream of events. 
+Also, each client can commit only the batches that were sent to him.
+
+The possible success answers from commit endpoints are:
+
+- `204`: cursors were successfully committed and offset was increased;
+- `200`: cursors were committed but at least one of the cursors didn't increase the offset as it was 
+less or equal to already committed one. In a case of this response code user will get a json in a 
+response body with a list of cursors and the results of their commits.
+
```

Comment:
@antban I think it makes sense to describe what @BGehrels mentioned, because it is not described anywhere. I'm not sure if it should be described here or not. But for now I will describe it here, and later we can move it to more appropriate place.